spec/requests/apis/unit_type_spec.rb

```diff
@@ -0,0 +1,103 @@
+
+      second_unit_type = create :unit_type,
+                                description: 'Apartamento de 2 quartos',
+                                condo: second_condo
+
+      third_unit_type = create :unit_type,
+                               description: 'Apartamento de 3 quartos',
+                               metreage: 90,
+                               fraction: 4,
+                               condo: first_condo
+
+      first_tower = create :tower, floor_quantity: 2,
+                                   units_per_floor: 2,
+                                   condo: first_condo
+
+      second_tower = create :tower,
+                            floor_quantity: 3,
+                            units_per_floor: 2,
+                            condo: second_condo
+
+      third_tower = create :tower,
+                           floor_quantity: 2,
+                           units_per_floor: 3,
+                           condo: first_condo
+
+      first_tower.generate_floors
+      second_tower.generate_floors
+      third_tower.generate_floors
+
+      first_tower.floors.each do |floor|
+        floor.units[0].update unit_type: first_unit_type
+        floor.units[1].update unit_type: third_unit_type
+      end
+
+      second_tower.floors.each do |floor|
+        floor.units[0].update unit_type: second_unit_type
+        floor.units[1].update unit_type: second_unit_type
+      end
+
+      third_tower.floors.each do |floor|
+        floor.units[0].update unit_type: third_unit_type
+        floor.units[1].update unit_type: third_unit_type
+        floor.units[2].update unit_type: first_unit_type
+      end
+
+      get '/api/v1/condos/1/unit_types'
+
+      expect(response).to have_http_status :ok
+      expect(response.parsed_body[0]['created_at']).not_to be_present
+      expect(response.parsed_body[0]['updated_at']).not_to be_present
+      expect(response.parsed_body.count).to eq 2
+
+      expect(response.parsed_body[0]['id']).to eq 1
+      expect(response.parsed_body[0]['description']).to eq 'Apartamento de 1 quarto'
+      expect(response.parsed_body[0]['metreage']).to eq '50.55'
+      expect(response.parsed_body[0]['fraction']).to eq '2.0'
+      expect(response.parsed_body[0]['unit_ids'].count).to eq 4
+
+      expect(response.parsed_body[1]['id']).to eq 3
+      expect(response.parsed_body[1]['description']).to eq 'Apartamento de 3 quartos'
+      expect(response.parsed_body[1]['metreage']).to eq '90.0'
+      expect(response.parsed_body[1]['fraction']).to eq '4.0'
+      expect(response.parsed_body[1]['unit_ids'].count).to eq 6
+    end
+
+    it "and there's no unit types" do
+      create :condo
+
+      get '/api/v1/condos/1/unit_types'
+
+      expect(response).to have_http_status :ok
+      expect(response.parsed_body).to be_empty
+    end
+
+    it 'and returns not found if condo is not found' do
+      get '/api/v1/condos/999999999/unit_types'
+
+      expect(response).to have_http_status :not_found
+    end
```

Comment:
Vocês utilizaram o status ok retornando vazio para quando não há tipos de unidades, mas not found para quando o condomínio não existe?
Eu pergunto isso por dúvida mesmo, ficamos em dúvida sobre retornar sucesso com perfil inexistente ou erro not found quando um CPF não é encontrado na nossa API.

Reply:
Isso, porque no primeiro caso, as rotas e os parâmetros estão corretos e retornou uma informação que existe (lista vazia de um condomínio que existe no banco de dados), já no segundo não, o condomínio 999999999 não existe, portanto o parâmetro é inválido.

No seu caso (validação de CPF) não tenho certeza, mas eu usaria o not found.